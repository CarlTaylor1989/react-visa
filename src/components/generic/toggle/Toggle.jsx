import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Translate } from 'react-redux-i18n';

import './Toggle.scss';

class Toggle extends Component {
  constructor(props) {
    super(props);
    this.onChange = this.onChange.bind(this);
  }

  onChange() {
    const { handleChange, checked } = this.props;
    handleChange(!checked);
  }

  render() {
    const { checked } = this.props;
    const className = checked ? ' checked' : '';
    return (
      <div className="toggleWrapper">
        <Translate className="toggleText" value="generic.switchOn" />
        <button
          className={`toggleSwitch${className}`}
          onClick={this.onChange}
          type="button"
        />
        <Translate className="toggleText" value="generic.switchOff" />
      </div>
    );
  }
}

Toggle.propTypes = {
  checked: PropTypes.bool.isRequired,
  handleChange: PropTypes.func.isRequired
};

export default Toggle;
