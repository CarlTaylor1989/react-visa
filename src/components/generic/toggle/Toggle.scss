@import '../../../styles/variables';
@import '../../../styles/mixins';

.toggleWrapper {
  @include align-items(center);
  display: flex;
  outline: none;

  .toggleSwitch {
    @include border-radius(26px);
    background: none;
    border: 1.5px solid $colour-white;
    box-sizing: border-box;
    cursor: pointer;
    height: 51px;
    margin: 0 12px;
    outline: none;
    position: relative;
    width: 102px;

    &::after {
      @include border-radius(16px);
      border: 1.5px solid $colour-white;
      box-sizing: border-box;
      content: '';
      display: block;
      height: 31px;
      left: 45px;
      position: absolute;
      top: 9px;
      transition: left .3s linear, top .3s linear;
      width: 43px;
    }

    &.checked {
      &::after {
        background: $colour-button;
        left: 9px;
      }
    }
  }

  .toggleText {
    font-size: 20px;
  }
}
