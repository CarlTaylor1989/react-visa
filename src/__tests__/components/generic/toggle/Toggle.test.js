/* global toJson */
import React from 'react';
import Toggle from '../../../../components/generic/toggle/Toggle';

let mockProps;
beforeEach(() => {
  jest.resetAllMocks();
  mockProps = {
    checked: false,
    handleChange: jest.fn()
  };
});

test('renders an unchecked Toggle', () => {
  const wrapper = shallow(<Toggle {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('renders a checked Toggle', () => {
  const wrapper = shallow(<Toggle {...mockProps} checked />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('method onChange() - sets the checked state', () => {
  const checked = true;
  const wrapper = shallow(<Toggle {...mockProps} />);
  wrapper.instance().onChange(checked);
  expect(mockProps.handleChange).toHaveBeenCalledWith(checked);
});
