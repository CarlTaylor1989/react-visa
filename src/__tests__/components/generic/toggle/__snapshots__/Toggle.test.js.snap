// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a checked Toggle 1`] = `
<div
  className="toggleWrapper"
>
  <Translate
    className="toggleText"
    tag="span"
    value="generic.switchOn"
  />
  <button
    className="toggleSwitch checked"
    onClick={[Function]}
    type="button"
  />
  <Translate
    className="toggleText"
    tag="span"
    value="generic.switchOff"
  />
</div>
`;

exports[`renders an unchecked Toggle 1`] = `
<div
  className="toggleWrapper"
>
  <Translate
    className="toggleText"
    tag="span"
    value="generic.switchOn"
  />
  <button
    className="toggleSwitch"
    onClick={[Function]}
    type="button"
  />
  <Translate
    className="toggleText"
    tag="span"
    value="generic.switchOff"
  />
</div>
`;
